nntrainer/nntrainer_error.h

```diff
@@ -24,13 +24,72 @@
 #define ML_ERROR_RESULT_OUT_OF_RANGE (-ERANGE)
 #endif
 
+#include <functional>
+#include <sstream>
 #include <stdexcept>
+
+#define NNTR_THROW_IF(pred, err) \
+  if (!(pred)) {                 \
```

Comment:
i wonder why `if(!pred) ... else ...` than just `if(pred) ...`?

Reply:
I think branch prediction would be the same or minimal as 99% or the time it would go to non-throw.
It is just a pure readability. I think it would be interesting to benchmark this though.

Reply:
Changed to `if((pred))`